server/runtime/src/main/java/org/infinispan/server/datasource/DataSourceFactory.java

```diff
@@ -46,6 +52,19 @@ public static DataSource create(DataSourceConfiguration configuration) throws SQ
             .leakTimeout(Duration.ofMillis(configuration.leakDetection()))
             .reapTimeout(Duration.ofMinutes(configuration.idleRemoval()));
 
+      if (configuration.driver().contains("postgresql")) {
+         pool.exceptionSorter(new PostgreSQLExceptionSorter());
+      } else if (configuration.driver().contains("mysql")) {
+         pool.exceptionSorter(new MySQLExceptionSorter());
```

Comment:
We can also use `MySQLExceptionSorter` for `mariadb` as they have the same error codes.

Reply:
Done